src/gui/layertree/qgslayertreeview.cpp

```diff
@@ -43,13 +44,22 @@ QgsLayerTreeView::QgsLayerTreeView( QWidget *parent )
   setEditTriggers( EditKeyPressed );
   setExpandsOnDoubleClick( false ); // normally used for other actions
 
+  // Ensure legend graphics are scrollable
+  header()->setStretchLastSection( false );
+  header()->setSectionResizeMode( QHeaderView::ResizeToContents );
+
+  // If vertically scrolling by item, legend graphics can get clipped
+  setVerticalScrollMode( QAbstractItemView::ScrollPerPixel );
```

Comment:
@nyalldawson, I'm thinking that if nothing else gets backported to `release-3_4`, this should, to fix vertical scrolling.

Reply:
@dakcarto I'm happy with that! (Also 3.8)